staging/src/k8s.io/client-go/rest/request_test.go

```diff
@@ -3712,3 +3712,335 @@ func TestRequestBodyResetOrder(t *testing.T) {
 		t.Errorf("Expected invocation request and response body operations for retry do not match: %s", cmp.Diff(expected, recorder.order))
 	}
 }
+
+func TestRetryableConditions(t *testing.T) {
+	methods := map[string]func(ctx context.Context, r *Request){
+		"Do": func(ctx context.Context, r *Request) {
+			r.Do(ctx)
+		},
+		"DoRaw": func(ctx context.Context, r *Request) {
+			r.DoRaw(ctx)
+		},
+		"Stream": func(ctx context.Context, r *Request) {
+			r.Stream(ctx)
+		},
+		"Watch": func(ctx context.Context, r *Request) {
+			w, err := r.Watch(ctx)
+			if err == nil {
+				// we need to wait here to avoid race condition.
+				<-w.ResultChan()
+			}
+		},
+	}
+
+	tests := []struct {
+		name             string
+		verb             string
```

Comment:
nit: consider turning verb into `verbs []string` to reduce the number of similar but slightly different test cases

Reply:
done, thanks